src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.EnumerateMatches.cs

```diff
@@ -149,6 +151,9 @@ public bool MoveNext()
                     return true;
                 }
 
+                _current = new ValueMatch(_regex.RightToLeft ? 0 : _input.Length, 0);
+                _startAt = _current.Index;
+                _prevLen = 0;
```

Comment:
I added this to improve performance if the user code suddenly calls MoveNext again after returning false. I know that user code usually doesn't do this.

Without this, each time MoveNext is called, the regular expression repeats the last failed search in the last segment of the string, which can be quite large. To avoid repeating the last failed search, I move the start of the next search to the end of the string. Accordingly, the last search section becomes empty.

Reply:
Calling MoveNext after it's already returned false is programmer error. It is not something we need to optimize for.